autoprecompiles/src/optimizer.rs

```diff
@@ -95,46 +88,14 @@ fn optimization_loop_iteration<T: FieldElement>(
     Ok(system)
 }
 
-pub fn optimize_pc_lookup<T: FieldElement>(
-    mut machine: SymbolicMachine<T>,
-    opcode: u32,
-    pc_lookup_bus_id: u64,
-) -> SymbolicMachine<T> {
-    let mut first_pc = None;
-    machine.bus_interactions.retain(|bus_int| {
-        if bus_int.id == pc_lookup_bus_id {
-            if first_pc.is_none() {
-                first_pc = Some(bus_int.clone());
-            }
-            return false;
-        }
-        true
-    });
-    let mut first_pc = first_pc.unwrap();
-    assert_eq!(first_pc.args.len(), 9);
-    first_pc.args[1] = AlgebraicExpression::Number(T::from(opcode));
-    first_pc.args[2] = AlgebraicExpression::Number(T::from(0u32));
-    first_pc.args[3] = AlgebraicExpression::Number(T::from(0u32));
-    first_pc.args[4] = AlgebraicExpression::Number(T::from(0u32));
-    first_pc.args[5] = AlgebraicExpression::Number(T::from(0u32));
-    first_pc.args[6] = AlgebraicExpression::Number(T::from(0u32));
-    first_pc.args[7] = AlgebraicExpression::Number(T::from(0u32));
-    first_pc.args[8] = AlgebraicExpression::Number(T::from(0u32));
-
-    machine.bus_interactions.push(first_pc);
-
-    machine
-}
-
 pub fn optimize_exec_bus<T: FieldElement>(
     mut machine: SymbolicMachine<T>,
     exec_bus_id: u64,
 ) -> SymbolicMachine<T> {
     let mut first_seen = false;
     let mut receive = true;
     let mut latest_send = None;
-    let mut subs_pc: BTreeMap<AlgebraicExpression<T>, AlgebraicExpression<T>> = Default::default();
```

Comment:
We previously assumed that execution bus interactions have as payload `(pc, timestamp)`. But there is no reason to assume that.